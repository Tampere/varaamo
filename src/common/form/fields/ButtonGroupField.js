import React from 'react';
import PropTypes from 'prop-types';
import FormGroup from 'react-bootstrap/lib/FormGroup';
import ControlLabel from 'react-bootstrap/lib/ControlLabel';
import ToggleButtonGroup from 'react-bootstrap/lib/ToggleButtonGroup';
import ToggleButton from 'react-bootstrap/lib/ToggleButton';

const ButtonGroupField = ({
  onChange,
  label,
  options,
  value,
  type,
  id,
}) => (
  <div className="app-ButtonGroupField">
    <FormGroup controlId={`buttonGroupField-${id}`}>
      {label && <ControlLabel className="app-ButtonGroupField__label">{label}</ControlLabel>}
      <ToggleButtonGroup
        className="app-ButtonGroupField__buttons"
        onChange={onChange}
        type={type}
        value={value}
      >
        {options.map(option => (
          <ToggleButton
            className="app-ButtonGroupField__button"
            key={`${id}-button-${option.value}`}
            value={option.value}
          >
            {option.label}
          </ToggleButton>
        ))}
      </ToggleButtonGroup>
    </FormGroup>
  </div>
);

const OptionShape = PropTypes.shape({
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  label: PropTypes.string.isRequired,
});

ButtonGroupField.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.array,
  ]),
  type: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(OptionShape),
};

export default ButtonGroupField;
