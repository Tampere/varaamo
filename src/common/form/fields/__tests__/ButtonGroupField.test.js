import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import ButtonGroupField from '../ButtonGroupField';

describe('ButtonGroupField', () => {
  test('renders correctly', () => {
    const props = {
      onChange: jest.fn(),
      label: 'foo',
      options: [
        { value: 'foo', label: 'Foo' },
        { value: 'bar', label: 'Bar' },
      ],
      value: 'foo',
      type: 'checkbox',
      id: 'foo',
    };

    const wrapper = shallow(
      <ButtonGroupField {...props} />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
