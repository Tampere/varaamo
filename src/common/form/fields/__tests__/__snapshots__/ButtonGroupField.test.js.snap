// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroupField renders correctly 1`] = `
<div
  className="app-ButtonGroupField"
>
  <FormGroup
    bsClass="form-group"
    controlId="buttonGroupField-foo"
  >
    <ControlLabel
      bsClass="control-label"
      className="app-ButtonGroupField__label"
      srOnly={false}
    >
      foo
    </ControlLabel>
    <Uncontrolled(ToggleButtonGroup)
      className="app-ButtonGroupField__buttons"
      onChange={[MockFunction]}
      type="checkbox"
      value="foo"
    >
      <ToggleButton
        className="app-ButtonGroupField__button"
        key="foo-button-foo"
        value="foo"
      >
        Foo
      </ToggleButton>
      <ToggleButton
        className="app-ButtonGroupField__button"
        key="foo-button-bar"
        value="bar"
      >
        Bar
      </ToggleButton>
    </Uncontrolled(ToggleButtonGroup)>
  </FormGroup>
</div>
`;
